cartography/intel/aws/sagemaker/models.py

```diff
@@ -0,0 +1,176 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.intel.aws.sagemaker.util import extract_bucket_name_from_s3_uri
+from cartography.models.aws.sagemaker.model import AWSSageMakerModelSchema
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_models(
+    boto3_session: boto3.session.Session,
+    region: str,
+) -> List[Dict[str, Any]]:
+    """
+    Get all SageMaker Models in the given region.
+    """
+    client = boto3_session.client("sagemaker", region_name=region)
+    paginator = client.get_paginator("list_models")
+    models: List[Dict[str, Any]] = []
+
+    # Get all model names
+    model_names: List[str] = []
+    for page in paginator.paginate():
+        for model in page.get("Models", []):
+            model_names.append(model["ModelName"])
+
+    # Get detailed information for each model
+    for model_name in model_names:
+        try:
```

Comment:
I don't think we need to have a `try+except` here because aws_handle_regions takes care of it.